src/java.desktop/share/classes/javax/swing/text/html/StyleSheet.java

```diff
@@ -1359,12 +1359,11 @@ private synchronized void getStyles(SelectorMapping parentMapping,
                            Vector<SelectorMapping> styles,
                            String[] tags, String[] ids, String[] classes,
                            int index, int numElements,
-                           Hashtable<SelectorMapping, SelectorMapping> alreadyChecked) {
-        // Avoid desending the same mapping twice.
-        if (alreadyChecked.contains(parentMapping)) {
+                           HashSet<SelectorMapping> alreadyChecked) {
+        // Avoid descending the same mapping twice.
+        if (!alreadyChecked.add(parentMapping)) {
```

Comment:
Clever!

Reply:
Seriously ? That is your explanation ? Do better.

Reply:
Hi Phil,
I'm not sure what you mean by this. I'd like to give feedback to client-related PRs when I can, and this line stood out to me as a smart change. I don't want to be critical all the time, so when I see a change I quite like, I want to let the person who wrote it know. Here, I thought the usage of `add(parentMapping)` was clever.